src/ui/src/js/configs/http_interceptor.js

```diff
@@ -39,33 +41,36 @@ export function authInterceptorService($q, $injector) {
         let $rootScope = $injector.get("$rootScope");
         let TokenService = $injector.get("TokenService");
 
+        let deferred = $q.defer();
+
         // This attempts to handle the condition where an access token has
         // expired but there's a refresh token in storage. We use the refresh
         // token to get a new access token then re-attempt the original request.
         let refreshToken = TokenService.getRefresh();
         if (refreshToken) {
-          return TokenService.doRefresh(refreshToken).then(
-            (response) => {
-              // Set the Authorization header to the updated default
-              rejection.config.headers.Authorization =
-                $http.defaults.headers.common.Authorization;
+          if (!inFlightAuthRequest) {
```

Comment:
This whole `inFlightAuthRequest` thing is to avoid attempting to refresh multiple times when there are multiple calls to the API on a single page and all of them result in a 401.  Without this, each call would result in its own refresh attempt, with them all basically stepping on each other because each refresh invalidates the preceding one.